test/forge/ReallocateWithdrawTest.sol

```diff
@@ -93,9 +102,21 @@ contract ReallocateWithdrawTest is IntegrationTest {
             withdrawnShares[2].toAssetsDown(totalSupplyAssets[2], totalSupplyShares[2])
         ];
 
-        if (withdrawnShares[0] > 0) withdrawn.push(MarketAllocation(allMarkets[0], 0, withdrawnShares[0]));
-        if (withdrawnAssets[1] > 0) withdrawn.push(MarketAllocation(allMarkets[1], withdrawnAssets[1], 0));
-        if (withdrawnShares[2] > 0) withdrawn.push(MarketAllocation(allMarkets[2], 0, withdrawnShares[2]));
+        if (withdrawnShares[0] > 0) {
+            withdrawn.push(MarketAllocation(allMarkets[0], 0, withdrawnShares[0]));
+            vm.expectEmit();
+            emit EventsLib.ReallocateWithdraw(allMarkets[0].id(), withdrawnAssets[0]);
+        }
+        if (withdrawnAssets[1] > 0) {
+            withdrawn.push(MarketAllocation(allMarkets[1], withdrawnAssets[1], 0));
+            vm.expectEmit();
+            emit EventsLib.ReallocateWithdraw(allMarkets[1].id(), withdrawnAssets[1]);
+        }
+        if (withdrawnShares[2] > 0) {
+            withdrawn.push(MarketAllocation(allMarkets[2], 0, withdrawnShares[2]));
+            vm.expectEmit();
+            emit EventsLib.ReallocateWithdraw(allMarkets[2].id(), withdrawnAssets[2]);
+        }
```

Comment:
```suggestion
        vm.expectEmit();
        if (withdrawnShares[0] > 0) {
            withdrawn.push(MarketAllocation(allMarkets[0], 0, withdrawnShares[0]));
            emit EventsLib.ReallocateWithdraw(allMarkets[0].id(), withdrawnAssets[0]);
        }
        if (withdrawnAssets[1] > 0) {
            withdrawn.push(MarketAllocation(allMarkets[1], withdrawnAssets[1], 0));
            emit EventsLib.ReallocateWithdraw(allMarkets[1].id(), withdrawnAssets[1]);
        }
        if (withdrawnShares[2] > 0) {
            withdrawn.push(MarketAllocation(allMarkets[2], 0, withdrawnShares[2]));
            emit EventsLib.ReallocateWithdraw(allMarkets[2].id(), withdrawnAssets[2]);
        }
```

1. did you try this?
2. it's weird to expect emit at this stage of the test. How would forge understand that we expect to emit in `reallocate`? Because it's the first non staticcall? Perhaps it works

Reply:
1. updating the PR
2. Because it's the first non staticcall? -> yes